tcs/tcsRunner.go

```diff
@@ -0,0 +1,118 @@
+	"github.com/multiversx/mx-multi-factor-auth-go-service/core"
+	"github.com/multiversx/mx-multi-factor-auth-go-service/factory"
+	storageFactory "github.com/multiversx/mx-multi-factor-auth-go-service/handlers/storage/factory"
+	"github.com/multiversx/mx-multi-factor-auth-go-service/metrics"
+	"github.com/multiversx/mx-sdk-go/authentication/native"
+	"github.com/multiversx/mx-sdk-go/core/http"
+)
+
+var log = logger.GetOrCreate("tcsRunner")
+
+type tcsRunner struct {
+	configs *config.Configs
+}
+
+// NewTcsRunner will create a new tcs runner instance
+func NewTcsRunner(cfgs *config.Configs) (*tcsRunner, error) {
+	if cfgs == nil {
+		return nil, ErrNilConfigs
+	}
+
+	return &tcsRunner{
+		configs: cfgs,
+	}, nil
+}
+
+// Start will trigger the tcs service
+func (tr *tcsRunner) Start() error {
+	cryptoComponents, err := factory.CreateCoreCryptoComponents()
+	if err != nil {
+		return err
+	}
+
+	statusMetricsHandler := metrics.NewStatusMetrics()
+	shardedStorageFactory := storageFactory.NewStorageWithIndexFactory(tr.configs.GeneralConfig, tr.configs.ExternalConfig, statusMetricsHandler)
+	registeredUsersDB, err := shardedStorageFactory.Create()
+	if err != nil {
+		return err
+	}
+
+	defer func() {
+		log.LogIfError(registeredUsersDB.Close())
+	}()
+
+	httpClient := http.NewHttpClientWrapper(nil, tr.configs.ExternalConfig.Api.NetworkAddress)
+	httpClientWrapper, err := core.NewHttpClientWrapper(httpClient)
+	if err != nil {
+		return err
+	}
+
+	twoFactorHandler, err := factory.CreateOTPHandler(tr.configs)
+	if err != nil {
+		return err
+	}
+
+	frozenOtpHandler, err := factory.CreateFrozenOTPHandler(tr.configs)
+	if err != nil {
+		return err
+	}
+
+	serviceResolver, err := factory.CreateServiceResolver(tr.configs, cryptoComponents, httpClientWrapper, registeredUsersDB, twoFactorHandler, frozenOtpHandler)
+	if err != nil {
+		return err
+	}
+
+	nativeAuthServerCacher, err := storageUnit.NewCache(tr.configs.GeneralConfig.NativeAuthServer.Cache)
+	if err != nil {
+		return err
+	}
+
+	tokenHandler := native.NewAuthTokenHandler()
+	args := native.ArgsNativeAuthServer{
+		HttpClientWrapper: httpClient,
+		TokenHandler:      tokenHandler,
+		Signer:            cryptoComponents.Singer(),
+		PubKeyConverter:   cryptoComponents.PubkeyConverter(),
+		KeyGenerator:      cryptoComponents.KeyGenerator(),
+		TimestampsCacher:  nativeAuthServerCacher,
+	}
+
+	nativeAuthServer, err := native.NewNativeAuthServer(args)
+	if err != nil {
+		return err
+	}
+
+	nativeAuthWhitelistHandler := middleware.NewNativeAuthWhitelistHandler(tr.configs.ApiRoutesConfig.APIPackages)
+
+	webServer, err := factory.StartWebServer(*tr.configs, serviceResolver, nativeAuthServer, tokenHandler, nativeAuthWhitelistHandler, statusMetricsHandler)
+	if err != nil {
+		return err
+	}
+
+	sigs := make(chan os.Signal, 1)
+	signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM)
+
+	<-sigs
+
+	log.Info("application closing, calling Close on all subcomponents...")
+
+	var lastErr error
```

Comment:
no need for this error with the current code, since there is only one error that can be returned

Reply:
right, removed it